Core/Core/Router/Router.swift

```diff
@@ -18,43 +18,40 @@
 
 import UIKit
 
-public struct RouteOptions: Equatable {
-    public let modal: Modal?
-    public let detail: Bool
-    public let embedInNav: Bool
-    public let addDoneButton: Bool
-
-    public struct Modal: Equatable {
-        public let presentationStyle: UIModalPresentationStyle?
-        public let inPresentation: Bool
-    }
-
-    public init(
-        modal: Modal? = nil,
-        detail: Bool = false,
+public enum RouteOptions: Equatable {
+    case push(addDoneButton: Bool = false)
+    case detail(embedInNav: Bool = false)
+    case modal(
+        _ style: UIModalPresentationStyle? = nil,
```

Comment:
I didn't realize swift lets you have defaults in enum cases, I thought we'd need a separate static func to handle default params. This is nice.